packages/marketplace/contracts/RoyaltiesRegistry.sol

```diff
@@ -200,40 +200,60 @@ contract RoyaltiesRegistry is OwnableUpgradeable, IRoyaltiesProvider {
     /// @return royalties 2981 royalty array
     function _getRoyaltiesEIP2981(address token, uint256 tokenId) internal view returns (Part[] memory royalties) {
         try IERC2981(token).royaltyInfo(tokenId, WEIGHT_VALUE) returns (address receiver, uint256 royaltyAmount) {
-            try IERC165Upgradeable(token).supportsInterface(type(IMultiRoyaltyRecipients).interfaceId) returns (
-                bool result
-            ) {
-                if (result) {
-                    try IMultiRoyaltyRecipients(token).getRecipients(tokenId) returns (
-                        Recipient[] memory multiRecipients
-                    ) {
-                        uint256 multiRecipientsLength = multiRecipients.length;
-                        royalties = new Part[](multiRecipientsLength);
-                        uint256 sum = 0;
-                        for (uint256 i; i < multiRecipientsLength; i++) {
-                            Recipient memory splitRecipient = multiRecipients[i];
-                            royalties[i].account = splitRecipient.recipient;
-                            uint256 splitAmount = (splitRecipient.bps * royaltyAmount) / WEIGHT_VALUE;
-                            royalties[i].value = splitAmount;
-                            sum += splitAmount;
-                        }
-                        // sum can be less than amount, otherwise small-value listings can break
-                        require(sum <= royaltyAmount, "RoyaltiesRegistry: Invalid split");
-                        return royalties;
-                    } catch {
-                        return _calculateRoyalties(receiver, royaltyAmount);
-                    }
-                } else {
-                    return _calculateRoyalties(receiver, royaltyAmount);
-                }
-            } catch {
+            if (_checkGetRecipientsInterface(token)) {
+                royalties = _getRecipients(token, tokenId, receiver, royaltyAmount);
```

Comment:
let's only use `return` OR only use the named return value (royalties)

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1240/commits/760b12fe94db15497c5e782e05b9b9360e78ee1f